lib/migrate-existing-resources.js

```diff
@@ -19,9 +23,28 @@ module.exports = function getCurrentState() {
         return this.getStackSummary(nestedStackId)
           .then(nestedStackSummary => {
             nestedStackSummary.forEach(nestedSummary => {
-              if (nestedSummary.LogicalResourceId in this.resourcesById) {
-                this.migrate(nestedSummary.LogicalResourceId, nestedStackName);
-              }
+              const logicalId = nestedSummary.LogicalResourceId;
+              const resource = this.resourcesById[logicalId];
+              if (!resource) return;
+              const migration = migrations[resource.Type];
+              if (!migration) return;
+
+              // Purpose of this step is to ensure that resources which were eventually distributed
+              // across suffixed stacks are in consequent deployments put into very same stacks
+              // (We don't want resources to be moved randomly between stacks across deployments)
+              if (!migration.allowSuffix) return;
+        
+              // Mark for migration only if resource was already deployed to stack as one
+              // resolved by stacks map rules
+              const destination = (typeof migration.destination === 'function')
+                ? migration.destination(logicalId, resource) : migration.destination;
+              if (!destination) return;
+
+              let stackNameBase = this.getStackNameBase(destination);
+              if (!nestedStackName.startsWith(stackNameBase)) return;
+              if (!isValidSuffix(nestedStackName.slice(stackNameBase.length))) return;
```

Comment:
I'm not sure I understand the scenario where this would be false, or if returning silently would be the best? Can you elaborate?

Reply:
Yes, e.g. we may have two different nested stacks `Foo` and `Foobar`, if we do not check suffix, we may accidentally acknowledge `Foobar` as suffixed `Foo`

Reply:
However, yes it's not right as `stackNameBase` will be in that case `FooNestedStack`, so this check is not doing what is was designed for. I'll remove it

Reply:
Suffix check is removed now